cloudinit/cmd/status.py

```diff
@@ -324,6 +328,35 @@ def _get_systemd_status() -> Optional[UXAppStatus]:
     return None
 
 
+def _get_error_or_running_from_systemd_with_retry(
+    existing_status, max_wait=5
+) -> Optional[UXAppStatus]:
+    """Get systemd status and retry if dbus isn't ready.
+
+    If cloud-init has determined that we're still running, then we can
+    ignore the status from systemd. However, if cloud-init has detected error,
+    then we should retry on systemd status so we don't incorrectly report
+    error state while cloud-init is still running.
+    """
+    start_time = time.time()
+    while time.time() - start_time < max_wait:
+        try:
+            return _get_error_or_running_from_systemd()
+        except subp.ProcessExecutionError:
+            if existing_status in (
+                UXAppStatus.DEGRADED_RUNNING,
+                UXAppStatus.RUNNING,
+            ):
+                return None
+            sleep(0.25)
+    print(
```

Comment:
I don't love printing here, but when trying to raise an exception, I didn't really find a better place higher in the call stack to catch and print, so here the print is.

Reply:
This works, we'll still be getting an exit 1 in this case indicating a failure anyway. I'd like to improve our message to better indicate the error raised by systemctl so if we have to debug this later we know the failure mode

```diff
--- a/cloudinit/cmd/status.py
+++ b/cloudinit/cmd/status.py
@@ -339,19 +339,21 @@ def _get_error_or_running_from_systemd_with_retry(
     error state while cloud-init is still running.
     """
     start_time = time.time()
+    last_error = None
     while time.time() - start_time < max_wait:
         try:
             return _get_error_or_running_from_systemd()
-        except subp.ProcessExecutionError:
+        except subp.ProcessExecutionError as e:
             if existing_status in (
                 UXAppStatus.DEGRADED_RUNNING,
                 UXAppStatus.RUNNING,
             ):
                 return None
+            last_error = e
             sleep(0.25)
     print(
         f"Failed to get status from systemd after {max_wait} seconds. "
-        "Cloud-init may still be running.",
+        f"Cloud-init may still be running. {last_error}",
         file=sys.stderr,
     )
     return None
```
Otherise +1

Reply:
thanks for the `{last_exception.stderr}` error text